### PR TITLE
fix/AB#82564_aggregate_cards_unnecessarily_reloading

### PR DESCRIPTION
**Pull request by RenzoPrats**
# Description

Fix: Now if some field of the filter is selected and this field is not in the widget filter, it's not reloaded the widget.

## Useful links

[ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/82564)

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

As the attached video.

## Screenshots

Peek 15-01-2024 16-05

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by LenaelleRelief** [comment]
I fixed a small bug that happened with maps but otherwise all good !

**Review by LenaelleL — APPROVED** [review]
I fixed a small bug that happened with maps but otherwise all good !

File: libs/shared/src/lib/services/context/context.service.ts [hunk 1]
```diff
@@ -530,4 +530,43 @@ export class ContextService {
...
+  /**
+   * Verify if changed filter is in widget filter
+   *
+   * @param obj1 previous filter object
+   * @param obj2 current filter object
+   * @param contextFilters widget context filter
+   *
+   * @returns different key
+   */
+  filterInWidgetFilter(obj1: any, obj2: any, contextFilters: any): boolean {
```
Comment (AntoineRelief) [inline libs/shared/src/lib/services/context/context.service.ts:566]: would have been better to call the arguments as 'previous' and 'current'

File: libs/shared/src/lib/services/context/context.service.ts [hunk 2]
```diff
@@ -530,4 +530,43 @@ export class ContextService {
...
+    const fieldFilterChanged = this.findFilterDifference(obj1, obj2);
```
Comment (AntoineRelief) [inline libs/shared/src/lib/services/context/context.service.ts:567]: being limited to a single field is certainly too restrictive

**Pull request closed without merging**